UNIXFS.md

```diff
@@ -90,7 +90,12 @@ UnixFS currently supports two optional metadata fields:
   - The remaining 20 bits are reserved for future use, and are subject to change. Spec implementations **MUST** handle bits they do not expect as follows:
     - For future-proofing the (de)serialization layer must preserve the entire uint32 value during clone/copy operations, modifying only bit values that have a well defined meaning: `clonedValue = ( modifiedBits & 07777 ) | ( originalValue & 0xFFFFF000 )`
     - Implementations of this spec must proactively mask off bits without a defined meaning in the implemented version of the spec: `interpretedValue = originalValue & 07777`
-* `mtime` -- The modification time in seconds since the epoch. This defaults to the unix epoch if unspecified
+* `mtime` -- The modification time in seconds since the epoch `1970-01-01T00:00:00Z`. The `repeated` protobuf type is overloaded to provide optional high resolution as follows:
+  - If no mtime value is available, an `undefined` mtime is assumed. Implementations are free to default to the epoch itself, or if appropriate render a variant of `N/A` to signify lack of a value
```

Comment:
I think the spec should have an opinion about default values.  It currently says:

> This defaults to the unix epoch if unspecified

Any reason not to do this?

Reply:
@achingbrain this is the same issue you had with `mode`: we need a way to represent "no mtime has been supplied". For FUSE-like bridges this would mean either `epoch` or `now() at time of mount`, and we could very well define this in spec. But for the web-gateway it should remain a `N/A` or `-` or something like that.

Reply:
`now() at time of mount` would probably need storing somewhere which might be problematic, probably saver to opt for `epoch`.

> But for the web-gateway it should remain a N/A or - or something like that.

Why is that?

Reply:
Because... `there is no mtime attached` != `Jan 1st 1970`?

I guess I am not understanding the question, can you clarify?

Reply:
So I guess we're saying something like:

> If the existence of `mtime` is important for your application, use `0` where it is not set

Seems fair enough.  Seeing `1970` everywhere when doing `ls -l` style mfs directory listings is bugging me a little.  I'd rather see `-` in it's place.

Reply:
I think the issue is that:

1. Operating systems expect _some_ mtime and we want to be consistent.
2. If we pick the epoch, the mtime will always be consistent. If we pick the mount time, it'll change every time we restart.

Reply:
On that basis we should probably stick with `0` as the default mtime as it is now.